### ISSUE TITLE
BlockDelimiters cop: Semantic block incorrectly recognised/autocorrected as procedural block

### ISSUE THREAD

**Issue by myabc** [bbatsov/rubocop#2021]
## Environment

`rubocop (0.32.1)`

with these settings:

```yaml
BlockDelimiters:
  Enabled: true
  EnforcedStyle: semantic
```

## Issue

To reproduce:

`rubocop -a` on the following code:

```ruby
      h[i] = [predecessor.detect { |j| j[key] == i },
              current.detect { |j| j[key] == i }]
```

will incorrectly flag an offense:

```
example.rb:1:30: C: [Corrected] Prefer do...end over {...} for procedural blocks.
  h[i] = [predecessor.detect { |j| j[key] == i },
```

and rewrite the code as such:

```ruby
      h[i] = [predecessor.detect do |j| j[key] == i end,
              current.detect { |j| j[key] == i }]
```

**Comment by myabc** [comment]
/ @NobodysNightmare
